src/syscall_mock/mod.rs

```diff
@@ -76,6 +76,24 @@ impl SyscallMock {
                     ptrace::setregs(pid, registers)?;
                 }
             }
+            (Syscall::Stat, SyscallStop::Exit) if !self.protocol.mocked_files.is_empty() => {
+                let filename = tracee_memory::peek_string(pid, registers.rdi)?;
+                if self.protocol.mocked_files.contains(&filename) {
+                    let buffer_ptr = registers.rsi;
```

Comment:
I think this variable name should include `stat`. Since that's the name of the struct. Maybe just `statbuf` like in the manpages?

Reply:
compromise on `statbuf_ptr`? i think `statbuf_ptr + file_mode_offset` is clearer if the name suggests it's a pointer.

Reply:
Sounds good to me.